### PR TITLE
[stable/sentry] Sentry filestore config

### PR DESCRIPTION
**Pull request by damoxc**
#### What this PR does / why we need it:

Adds support for configuring Sentry's file storage.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by damoxc** [comment]
Was there something I missed? Seems mad to have to maintain either a fork of the container image or the chart to use the GCS filestore backend.

**Comment by viglesiasce** [comment]
Apologies that I didn't look at this sooner

**Comment by viglesiasce** [comment]
@damoxc can you take a look at the errors here. I'm not sure if those are expected or not:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/helm_charts/11153/pull-charts-e2e/21510/

**Comment by damoxc** [comment]
@viglesiasce sure thing, I'll try and take a look in the next few days. Kinda looks like the database hasn't been set up correctly which I wouldn't have thought would be related but you never know...

**Comment by damoxc** [comment]
Looking at the results from https://github.com/helm/charts/commit/5065067a9dc8f02426559e1ebf74c0c21469d0ff (https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/helm_charts/12826/pull-charts-e2e/1117765663113351172/build-log.txt) the errors from db-init seem to occur as well, suggesting these changes aren't introducing the errors at least.

**Review by guyromb — APPROVED** [review]

**Comment by gboor** [comment]
Can we get this merged? I really need this and there seems to be no reason not to merge it...

**Comment by paulczar** [comment]
Yeah lets get this merged!  If you can clean up the merge conflicts I'll lgtm it.  feel free to hit me up on slack when you do it to jog my memory.

**Comment by gboor** [comment]
@paulczar this is not my merge request, so there's not much I can do to clean up anything. Hopefully @damoxc is still around?

**Comment by damoxc** [comment]
@gboor @paulczar how does that look?

**Comment by gboor** [comment]
@damoxc looks great to me! But I'm just an end-user in this particular case :P

**Comment by maorfr** [comment]
/assign @okgolove 

@damoxc, if you wish to get this merged, i'm sure @okgolove will be willing to review this!

File: stable/sentry/values.yaml [hunk 1]
```diff
@@ -109,39 +109,52 @@ service:
...
+      ## different nodes.
+      persistentWorkers: false
 
-  # Where to store sentry files
-  # https://docs.sentry.io/server/filestore/
-  filestore_dir: /var/lib/sentry/files
+  ## Point this at a pre-configured secret containing a service account. The resulting
+  ## secret will be mounted at /var/secrets/google
+  gcs:
+    credentialsFile: credentials.json
```
Comment (okgolove) [inline stable/sentry/values.yaml:150]: Is /var/secrets is a GKE special folder? I've never used it.

**Comment by okgolove** [comment]
Thank you @maorfr.

Hi @damoxc. Please, rebase the branch.

Also can you tell me a bit about /var/secrets/ folder?

**Comment by damoxc** [comment]
@okgolove the `/var/secrets` folder was simply a location to place the key, and looking at it, I think I intended `/var/run/secrets` to match where Kubernetes places the default service account key.

I've added a commit changing this. Once you're happy I can rebase this into a single commit.

**Comment by damoxc** [comment]
@okgolove would you still like it rebased into one commit?

**Comment by okgolove** [comment]
@damoxc you don't need to do it. Prow will squash them automatically.

**Review by okgolove — APPROVED** [review]

File: stable/sentry/values.yaml (hunk 1 above)
Comment (damoxc) [reply stable/sentry/values.yaml:150]: Whoops, missed that when I updated it, should be `/var/run/secrets` now. Updated.

**Comment by okgolove** [comment]
I think this thing needs to be debugged.
@damoxc do you have any problems during deploy the chart?

**Comment by damoxc** [comment]
@okgolove I didn't in the initial revision, but the changes made to the chart since opening this PR may have caused some problems, I'll have a look.

**Comment by damoxc** [comment]
@okgolove tests look to be OK now

**Comment by kevinyan** [comment]
How was this problem finally solved? I seem to have asked a similar question, which has troubled me for a long time. Help God to see

[Issue:Unknown error 524 when upload the sourcemaps](https://github.com/getsentry/sentry/issues/16462)

**Pull request merged**